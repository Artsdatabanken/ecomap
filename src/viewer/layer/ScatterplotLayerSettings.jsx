import React from 'react'
import ActionAllOut from 'material-ui/svg-icons/action/all-out'
import SliderSetting from './SliderSetting'

export default class HexagonLayerSettings extends React.Component {
  render () {
    const {radius, onChange} = this.props
    return (<div>
      <SliderSetting
        title='Radius'
        value={radius}
        icon={<ActionAllOut />}
        onChange={value => onChange('radius', value)} />
    </div>)
  }
}
