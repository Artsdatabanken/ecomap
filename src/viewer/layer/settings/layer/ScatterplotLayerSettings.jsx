import React from "react";
import ActionAllOut from "material-ui/svg-icons/action/all-out";
import SliderSetting from "../SliderSetting";
import ColorpickerSetting from '../ColorpickerSetting'

export default class ScatterplotLayerSettings extends React.Component {
  render() {
    const { radius, onChange, fillColor, fillOpacity } = this.props;
    return (
      <div>
        <SliderSetting
          title="Radius"
          value={radius}
          icon={<ActionAllOut />}
          onChange={value => onChange("radius", value)}
        />
        <ColorpickerSetting
          onChange={onChange}
          fillColor={fillColor}
          fillOpacity={fillOpacity}
        />
      </div>
    );
  }
}
