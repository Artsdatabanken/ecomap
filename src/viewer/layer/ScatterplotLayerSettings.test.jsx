import React from 'react'
import renderer from 'react-test-renderer'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import muiTheme from '../../theme_jest'

import ScatterplotLayerSettings from './ScatterplotLayerSettings'

it('loading standard tile', () => {
  const tree = renderer.create(
    <MuiThemeProvider muiTheme={muiTheme}>
      <ScatterplotLayerSettings fillOpacity={0.1} radius={0.2} />
    </MuiThemeProvider>
      )
  expect(tree.toJSON()).toMatchSnapshot()
})
