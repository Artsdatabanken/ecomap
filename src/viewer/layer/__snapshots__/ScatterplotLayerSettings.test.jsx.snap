// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading standard tile 1`] = `
<div>
  <div>
    <div
      style={
        Object {
          "display": "inline-block",
          "height": "58px",
          "marginTop": "8px",
        }
      }
    >
      <div
        style={
          Object {
            "float": "left",
            "position": "relative",
            "top": "21px",
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "MozTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "MozUserSelect": "none",
              "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "WebkitUserSelect": "none",
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "msUserSelect": "none",
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M16.21 4.16l4 4v-4zm4 12l-4 4h4zm-12 4l-4-4v4zm-4-12l4-4h-4zm12.95-.95c-2.73-2.73-7.17-2.73-9.9 0s-2.73 7.17 0 9.9 7.17 2.73 9.9 0 2.73-7.16 0-9.9zm-1.1 8.8c-2.13 2.13-5.57 2.13-7.7 0s-2.13-5.57 0-7.7 5.57-2.13 7.7 0 2.13 5.57 0 7.7z"
          />
        </svg>
      </div>
      <div
        style={
          Object {
            "float": "left",
            "height": "48px",
            "paddingLeft": "16px",
            "position": "relative",
            "top": "0px",
            "width": "310px",
          }
        }
      >
        <div
          className={undefined}
          style={
            Object {
              "MozTransition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "WebkitTransition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": 256,
            }
          }
        >
          <label
            className={undefined}
            htmlFor="undefined-undefined-Radius-63915"
            onTouchTap={undefined}
            style={
              Object {
                "MozTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "MozUserSelect": "none",
                "WebkitTransform": "scale(0.75) translate(0, -28px)",
                "WebkitTransformOrigin": "left top",
                "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "WebkitUserSelect": "none",
                "color": "rgba(0, 0, 0, 0.3)",
                "left": "-40px",
                "lineHeight": "22px",
                "msTransform": "scale(0.75) translate(0, -28px)",
                "msTransformOrigin": "left top",
                "msUserSelect": "none",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": "22px",
                "transform": "scale(0.75) translate(0, -28px)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }
          >
            Radius
          </label>
          <div
            id="undefined-undefined-Radius-63915"
            style={
              Object {
                "MozBoxSizing": "border-box",
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
          >
            <div
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "MozUserSelect": "none",
                  "WebkitUserSelect": "none",
                  "cursor": "default",
                  "height": 18,
                  "marginBottom": 48,
                  "marginTop": 24,
                  "msUserSelect": "none",
                  "muiPrepared": true,
                  "position": "relative",
                  "touchCallout": "none",
                  "userSelect": "none",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "height": 2,
                    "left": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "top": 8,
                    "width": "100%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "MozTransition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "WebkitTransition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "[object Object]": 0,
                      "backgroundColor": "#00bcd4",
                      "directionInvariant": true,
                      "height": "100%",
                      "left": 0,
                      "marginRight": 6,
                      "muiPrepared": true,
                      "position": "absolute",
                      "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": Array [
                        "-webkit-calc(20%)",
                        "-moz-calc(20%)",
                        "calc(20%)",
                      ],
                    }
                  }
                />
                <div
                  style={
                    Object {
                      "MozTransition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "WebkitTransition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "[object Object]": 0,
                      "backgroundColor": "#bdbdbd",
                      "directionInvariant": true,
                      "height": "100%",
                      "marginLeft": 6,
                      "muiPrepared": true,
                      "position": "absolute",
                      "right": 0,
                      "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": Array [
                        "-webkit-calc(80%)",
                        "-moz-calc(80%)",
                        "calc(80%)",
                      ],
                    }
                  }
                />
                <div
                  style={
                    Object {
                      "MozBoxSizing": "border-box",
                      "MozTransition": "background 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, border-color 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, width 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, height 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "WebkitTransform": "translate(-50%, -50%)",
                      "WebkitTransition": "background 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, border-color 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, width 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, height 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "backgroundClip": "padding-box",
                      "backgroundColor": "#00bcd4",
                      "border": "0px solid transparent",
                      "borderRadius": "50%",
                      "boxSizing": "border-box",
                      "cursor": "pointer",
                      "directionInvariant": true,
                      "height": 12,
                      "left": "20%",
                      "margin": "1px 0 0 0",
                      "msTransform": "translate(-50%, -50%)",
                      "muiPrepared": true,
                      "outline": "none",
                      "overflow": "visible",
                      "pointerEvents": "inherit",
                      "position": "absolute",
                      "top": 0,
                      "transform": "translate(-50%, -50%)",
                      "transition": "background 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, border-color 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, width 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, height 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": 12,
                      "zIndex": 1,
                    }
                  }
                  tabIndex={0}
                >
                  <div
                    style={
                      Object {
                        "height": 12,
                        "left": 0,
                        "muiPrepared": true,
                        "overflow": "visible",
                        "position": "absolute",
                        "top": 0,
                        "width": 12,
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <input
              max={1}
              min={0}
              name={undefined}
              required={true}
              step={0.05}
              type="hidden"
              value={0.2}
            />
          </div>
        </div>
      </div>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
  </div>
</div>
`;
